src/System.Private.CoreLib/shared/System/Globalization/CalendarData.Unix.cs

```diff
@@ -366,36 +373,36 @@ private static bool EnumMonthNames(string localeName, CalendarId calendarId, Cal
                 if (callbackContext.Results.Count > 13)
                 {
                     Debug.Assert(calendarId == CalendarId.HEBREW && callbackContext.Results.Count == 14);
-                    
+
                     if (calendarId == CalendarId.HEBREW)
                     {
                         leapHebrewMonthName = callbackContext.Results[13];
                     }
                     callbackContext.Results.RemoveRange(13, callbackContext.Results.Count - 13);
-                }                
+                }
 
                 monthNames = callbackContext.Results.ToArray();
             }
 
             return result;
         }
 
-        private static bool EnumEraNames(string localeName, CalendarId calendarId, CalendarDataType dataType, out string[] eraNames)
+        private static bool EnumEraNames(string localeName, CalendarId calendarId, CalendarDataType dataType, out string[]? eraNames)
         {
             bool result = EnumCalendarInfo(localeName, calendarId, dataType, out eraNames);
 
             // .NET expects that only the Japanese calendars have more than 1 era.
             // So for other calendars, only return the latest era.
-            if (calendarId != CalendarId.JAPAN && calendarId != CalendarId.JAPANESELUNISOLAR && eraNames.Length > 0)
+            if (calendarId != CalendarId.JAPAN && calendarId != CalendarId.JAPANESELUNISOLAR && eraNames?.Length > 0)
```

Comment:
note: IL will change a bit here (null check) - I wasn't able to prove this is not nullable by annotating although at the same time Japan is not expected to not have era names

Reply:
You could add a Debug.Assert before this line, `eraNames != null` that should get rid of the warning, no?